src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -111,7 +154,11 @@ private function completeDefinition($id, Definition $definition)
                 // Typehint against a non-existing class
 
                 if (!$parameter->isDefaultValueAvailable()) {
-                    throw new RuntimeException(sprintf('Cannot autowire argument %s for %s because the type-hinted class does not exist (%s).', $index + 1, $definition->getClass(), $reflectionException->getMessage()), 0, $reflectionException);
+                    if ($isConstructor) {
+                        throw new RuntimeException(sprintf('Cannot autowire argument %s for %s because the type-hinted class does not exist (%s).', $index + 1, $definition->getClass(), $reflectionException->getMessage()), 0, $reflectionException);
+                    }
+
+                    return;
```

Comment:
Return is the good keyword here IMO. The idea is to stop because this method is not autowirable:

* if it's a constructor, throw an exception
* if it's a setter, return and ignore this setter